### PR TITLE
[WinForms] Matching Mono PropertyGrid behaviour with .Net one

### LINKED ISSUE #11617: [WinForms] PropertyGrid and custom types converting issues
There are two issues related with custom type converting:
1. `PropertyGrid` partially ignores custom `TypeConverter` assigned to a property. I mean, that in .Net on Windows such `TypeConverter` allows to override `GetPropertiesSupported` and `GetProperties` methods. One may use this to add custom properties (i.e. custom child grid entries) to an attributed property. 
2. `ITypeDescriptorContext` interface by default points to a parent `GridEntry`, not the current one. This leads to an obvious problem: any `TypeConverter.GetPropertiesSupported` and `TypeConverter.GetProperties` get incorrect context. The `ITypeDescriptorContext.Instance` and refers `ITypeDescriptorContext.PropertyDescriptor` to a parent object and its property. The small attached program [test_PropertyGrid.zip](https://github.com/mono/mono/files/2562637/test_PropertyGrid.zip) illustrate this issue.

### PR DESCRIPTION
**Pull request by NVoronchev**
Fixes #11617

### CODE REVIEW

File: mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs [hunk 1]
```diff
@@ -249,12 +249,11 @@ public override bool Select ()
 		}
 
 		#region ITypeDescriptorContext
-		void ITypeDescriptorContext.OnComponentChanged () 
-		{
+
+		void ITypeDescriptorContext.OnComponentChanged () {
```
Comment (luhenry) [inline mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs]: Reduce line damage. Revert that change.

File: mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs [hunk 2]
```diff
@@ -249,12 +249,11 @@ public override bool Select ()
...
 		}
 
-		bool ITypeDescriptorContext.OnComponentChanging () 
-		{
+		bool ITypeDescriptorContext.OnComponentChanging () {
```
Comment (luhenry) [inline mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs]: Reduce line damage. Revert that change.

File: mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs [hunk 3]
```diff
@@ -271,20 +270,13 @@ bool ITypeDescriptorContext.OnComponentChanging ()
...
 
 		PropertyDescriptor ITypeDescriptorContext.PropertyDescriptor {
-			get {
-				if (ParentEntry != null && ParentEntry.PropertyDescriptor != null)
-					return ParentEntry.PropertyDescriptor; 
-				return PropertyDescriptor;
-			}
+			get { return PropertyDescriptor; }
 		}
+
```
Comment (luhenry) [inline mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs]: Reduce line damage. Revert that change.

File: mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs [hunk 4]
```diff
@@ -792,7 +784,7 @@ private bool IsPropertyMergeable (PropertyDescriptor property)
 		{
 			if (property == null)
 				return false;
-				
+
```
Comment (luhenry) [inline mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs]: Reduce line damage. Revert that change.

File: mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs [hunk 5]
```diff
@@ -827,7 +819,7 @@ private string[] GetMergedPropertyNames (object [] objects)
 			string[] propertyNames = new string [intersection.Count];
 			for (int i=0; i < intersection.Count; i++)
 				propertyNames[i] = ((PropertyDescriptor)intersection[i]).Name;
-				
+
```
Comment (luhenry) [inline mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs]: Reduce line damage. Revert that change.

**Comment by luhenry** [comment]
@monojenkins build

File: mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs (hunk 4 above)
Comment (NVoronchev) [reply mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs]: @luhenry, I guess this change does not damage line number. I have removed meaningless tab symbols without line number changes. Totally empty lines (without any kind of spaces) is a normal condition for the current file. IMHO, of course :)

File: mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs (hunk 5 above)
Comment (NVoronchev) [reply mcs/class/System.Windows.Forms/System.Windows.Forms/GridEntry.cs]: same comment, as https://github.com/mono/mono/pull/11618/commits/32f310893f277ab4d1fca1bc69744c0ac30d1969#r234181420

**Comment by luhenry** [comment]
@monojenkins build failed

**Pull request merged**